src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpHandlers.java

```diff
@@ -0,0 +1,168 @@
+     *
+     * <p> This method creates a <i>handleOrElse</i> handler; an if-else like
+     * construct. Exchanges who's request matches the {@code handlerTest}
+     * predicate are handled by the {@code handler}. All remaining exchanges
+     * are handled by the {@code fallbackHandler}.
+     *
+     * <p> Example of a nested handleOrElse handler:
+     * <pre>{@code
+     *    Predicate<Request> IS_GET = r -> r.getRequestMethod().equals("GET");
+     *    Predicate<Request> WANTS_DIGEST =  r -> r.getRequestHeaders().containsKey("Want-Digest");
+     *
+     *    var h1 = new SomeHandler();
+     *    var h2 = HttpHandlers.handleOrElse(IS_GET, new SomeGetHandler(), h1);
+     *    var h3 = HttpHandlers.handleOrElse(WANTS_DIGEST.and(IS_GET), new SomeDigestHandler(), h2);
+     * }</pre>
+     * The {@code h3} handleOrElse handler delegates handling of the exchange to
+     * {@code SomeDigestHandler} if the "Want-Digest" request header is present
+     * and the request method is {@code GET}, otherwise it delegates handling of
+     * the exchange to the {@code h2} handler. The {@code h2} handleOrElse
+     * handler, in turn, delegates handling of the exchange to {@code
+     * SomeGetHandler} if the request method is {@code GET}, otherwise it
+     * delegates handling of the exchange to the {@code h1} handler. The {@code
+     * h1} handler handles all exchanges that are not previously delegated to
+     * either {@code SomeGetHandler} or {@code SomeDigestHandler}.
+     *
+     * @param handlerTest a request predicate
+     * @param handler a conditional handler
+     * @param fallbackHandler a fallback handler
+     * @return a handler
+     * @throws NullPointerException if any argument is null
+     */
+    public static HttpHandler handleOrElse(Predicate<Request> handlerTest,
+                                           HttpHandler handler,
+                                           HttpHandler fallbackHandler) {
+        Objects.requireNonNull(handlerTest);
+        Objects.requireNonNull(handler);
+        Objects.requireNonNull(fallbackHandler);
+        return exchange -> {
+            if (handlerTest.test(exchange))
+                handler.handle(exchange);
+            else
+                fallbackHandler.handle(exchange);
+        };
+    }
+
+    /**
+     * Returns an {@code HttpHandler} that sends a response comprising the given
+     * {@code statusCode}, {@code headers}, and {@code body}.
+     *
+     * <p> This method creates a handler that reads and discards the request
+     * body before it sets the response state and sends the response.
+     *
+     * <p> {@code headers} are the effective headers of the response. The
+     * response <i>body bytes</i> are a {@code UTF-8} encoded byte sequence of
+     * {@code body}. The response {@linkplain HttpExchange#sendResponseHeaders(int, long) is sent}
+     * with the given {@code statusCode} and the body bytes' length. The body
```

Comment:
That might give the impression that chunked encoding will be used if the body length is 0. I wonder ifit should say instead:

```
with the given {@code statusCode} and a {@code Content-Length} field set to the body bytes' length.
```

Reply:
Or maybe - which would be more accurate:

```
with the given {@code statusCode} and the body bytes' length (or {@code -1} if the body is empty).
```

Reply:
I agree with your second suggestion. It's better not to refer to the 'Content-Length' header at all.

Reply:
Indeed more accurate, I'll update to the second suggestion.